libbeat/cmd/instance/beat.go

```diff
@@ -984,6 +985,64 @@ func (b *Beat) createOutput(stats outputs.Observer, cfg config.Namespace) (outpu
 		return outputs.Group{}, nil
 	}
 
+	// TODO (Tiago): find a proper name for this logger
+	logger := logp.L().Named("ssl.cert.reloader")
+	// Here the output is created and we have acces to the Beat struct (with the manager)
+	// as a workaround we can unpack the new settings and trigger the reload-watcher from here
+
+	// We get an output config, so we extract the 'SSL' bit from it
+	rawTLSCfg, err := cfg.Config().Child("ssl", -1)
+	if err != nil {
+		return outputs.Group{}, fmt.Errorf("could not extract the 'ssl' section of the output config: %w", err)
+	}
+
+	tlsCfg := tlscommon.Config{}
+	if err := rawTLSCfg.Unpack(&tlsCfg); err != nil {
+		logger.Errorf("could not unpack config: %s", err)
+		return outputs.Group{}, fmt.Errorf("unpacking 'ssl' config: %w", err)
+	}
+
+	watchers := make([]*cfgfile.GlobWatcher, 0, len(tlsCfg.CAs))
+	for _, ca := range tlsCfg.CAs {
+		if tlscommon.IsPEMString(ca) {
+			// That's an embedded cert, we're only interested in files
+			continue
+		}
+
+		logger.Debugf("watching '%s' for changes", ca)
+		gw := cfgfile.NewGlobWatcher(ca)
```

Comment:
this may or may not be an issue.  But the CA from the config point of view is a filepath, not a glob.  So if the filepath has something *, ?, [, ], or \\ in it we will get "interesting" results.

Reply:
That's a good one, thanks!. I was wondering if I could use the filepath as a glob. On a quick look I did not find a "FileWatcher", so I went with the Glob one to prove the concept/get an idea of how the code will look like. I'll make a note to look into this later.

I wonder who would use any of those characters on a filename on a server... However we do need to cover it.

Reply:
I know it is an edge case, but then who would name their kid ["Robert'); DROP Table Students;--"](https://xkcd.com/327/) ? 😃

Reply:
I love his one, hauahuahuahua

But for some reason I always thought his name was Bob. Or maybe I'm just too used  to Alice, Bob and Eve :P